tests_and_analysis/test/euphonic_test/test_broadening.py

```diff
@@ -0,0 +1,76 @@
+import pytest
+import numpy as np
+import numpy.testing as npt
+from scipy.integrate import simps
+from scipy.stats import norm
+
+from euphonic.broadening import find_coeffs, gaussian, variable_width
+from euphonic import ureg
+from tests_and_analysis.test.euphonic_test.test_force_constants\
+    import get_fc_path
+from tests_and_analysis.test.euphonic_test.test_qpoint_frequencies\
+    import get_qpt_freqs
+from tests_and_analysis.test.utils import get_mode_widths
+from tests_and_analysis.test.euphonic_test.test_spectrum1d\
+    import get_expected_spectrum1d
+
+
+@pytest.mark.parametrize(
+        ('material, qpt_freqs_json, mode_widths_json, ebins'), [
+            ('quartz', 'quartz_554_full_qpoint_frequencies.json',
+             'quartz_554_full_mode_widths.json',
+             np.arange(0, 155, 0.1)*ureg('meV'))])
+def test_area_unchanged_for_broadened_dos(material, qpt_freqs_json,
+                                          mode_widths_json, ebins):
+    """
+    Test that the area is approximately equal for unbroadened and
+    broadened dos
+    """
+    qpt_freqs = get_qpt_freqs(material, qpt_freqs_json)
+    mode_widths = get_mode_widths(get_fc_path(mode_widths_json))
+    dos = qpt_freqs.calculate_dos(ebins)
+    weights = np.ones(qpt_freqs.frequencies.shape) * \
+        np.full(qpt_freqs.n_qpts, 1/qpt_freqs.n_qpts)[:, np.newaxis]
+    variable_width_broaden = variable_width(ebins, qpt_freqs.frequencies,
+                                            mode_widths, weights,
+                                            0.01)
+    ebins_centres = ebins.magnitude[:-1] + 0.5*np.diff(ebins.magnitude)
+    dos_area = simps(dos.y_data, ebins_centres)
+    adaptively_broadened_dos_area = simps(variable_width_broaden,
+                                          ebins_centres)
+    assert adaptively_broadened_dos_area == pytest.approx(dos_area, rel=0.01)
+
+@pytest.mark.parametrize(
+        ('material, qpt_freqs_json, mode_widths_json,'
+         'expected_dos_json, ebins'), [
+            ('quartz', 'toy_quartz_qpoint_frequencies.json',
+             'toy_quartz_mode_widths.json',
+             'toy_quartz_adaptive_dos.json',
+             np.arange(0, 40, 0.1)*ureg('meV'))])
+def test_lower_bound_widths_broadened(material, qpt_freqs_json,
```

Comment:
Ah I see, good spot! I think because this set of inputs needs to be tested specifically to prevent the bug reoccurring it's fine to have it as a separate test, maybe just add a comment in the test describing the bug?

Reply:
Okay will do!